### PR TITLE
Tools: Add prerequisites install script for openSUSE Tumbleweed

### PR DESCRIPTION
**Pull request by Jaaaky**
Tried to make it as complete as compatible as possible with Ubuntu environment

Installing static libs for SITL_static
And gcc_arm-linux for Linux based boards

### CODE REVIEW

**Comment by magicrub** [comment]
Is tumbleweed a version of SUSE? If so, it shouldn't be in the filename but instead just be handled within the file for different versions

**Review by magicrub — COMMENTED** [review]
Minor spelling issue

**Comment by Jaaaky** [comment]
> Is tumbleweed a version of SUSE? If so, it shouldn't be in the filename but instead just be handled within the file for different versions

No it's actually a rolling release flavor - there are major SUSE versions:
1. SUSE Enterprise for commercial support and so on, think of it a RHEL
2. openSUSE leap that's stable, new release every many months, think of it as ubuntu LTS
3. openSUSE tumbleweed which is rolling release distribution where a new version is released every few days, much like Acrhlinux/Manjaro and so on.

So for leap handling it'd be another story where tracking of versions and so on would occur within the file itself much like with ubuntu script.

**Comment by Jaaaky** [comment]
> Minor spelling issue

Sorry, where is it? I cannot find your comment, maybe it has gone before the force-update?

**Comment by Jaaaky** [comment]
@magicrub 
This pr is ready for merge. I've tested on different setups including WSL installations.
ccache config is approved by @tridge

File: Tools/environment_install/install-prereqs-openSUSE-Tumbleweed.sh [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+if $ASSUME_YES; then
+    ZYPPER="sudo zypper in -y --no-recommends"
+fi
+PIP3=pip3
+if $QUIET; then
+    PIP3="pip3 -q"
+fi
+
+function package_is_installed() {
+    rpm -q ModemManager &>/dev/null
```
Comment (peterbarker) [inline Tools/environment_install/install-prereqs-openSUSE-Tumbleweed.sh:44]: Ahhh... is that the *only* package that TumbleWeed offers?

File: Tools/environment_install/install-prereqs-openSUSE-Tumbleweed.sh [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    fi
+fi
+
+$PIP3 install -U pip setuptools wheel
+$PIP3 install -U attrdict3
+$PIP3 install -U $PYTHON_PKGS
+
+(
+    cd /usr/lib64/ccache
+    if [ ! -f arm-none-eabi-g++ ]; then
```
Comment (peterbarker) [inline Tools/environment_install/install-prereqs-openSUSE-Tumbleweed.sh:114]: Looks like a for loop is waiting to be born here.

Probably also in the source script....

**Review by peterbarker — CHANGES_REQUESTED** [review]
Is there some stuff we could factor out here somewhere?  Looks like there's a chunk of stuff copied which could be shared.

I don't think it needs to be done here, but anything you share with other scripts is stuff which someone may else fix when it breaks...

**Comment by Jaaaky** [comment]
@peterbarker 
All done/fixed and re-tested.
Thank you
For other scripts specially for Ubuntu as it's more widely used and I do not personally use it, it's better to be modified by their maintainer.

File: Tools/environment_install/install-prereqs-openSUSE-Tumbleweed.sh (hunk 2 above)
Comment (Jaaaky) [reply Tools/environment_install/install-prereqs-openSUSE-Tumbleweed.sh]: done for this script only

**Review by peterbarker — APPROVED** [review]
I've squashed this into one commit and will mark it MergeOnCIPass

**Pull request merged**